compiler/rustc_hir_typeck/src/lib.rs

```diff
@@ -221,12 +221,6 @@ fn typeck_with_fallback<'tcx>(
             }))
         } else if let Node::AnonConst(_) = node {
             match tcx.hir().get(tcx.hir().parent_id(id)) {
-                Node::Expr(&hir::Expr {
```

Comment:
would be nice to assert that we're only calling typeck_with_fallback on things that aren't typeck children, just in case someone messes with that `typeck_root_def_id != def_id` branch above, but whatever